src/Python/measurement_unit.py

```diff
@@ -0,0 +1,351 @@
+# -*- coding: utf-8 -*-
+"""
+MeasurementUnit
+
+"""
+
+from scipy.constants import F2C, K2C, C2F, C2K
+
+def C2C(x):
+    """
+    An identity function, to convert Celsius to Celsius
+
+    """
+    return x
+
+
+class MeasurementUnit():
+    """
+    Encapsulates the notion of a measurement unit, with an optional SI prefix.
+    
+    Provides methods to convert to and from the canonical representation:
+        Temporal data: seconds ('s')
+        Spatial data: millimetres ('mm')
+        Temperature data: degrees Celsius ('C')
+        Dimensionless data: (no units) ('')
+
+    Usage
+    -----------
+    
+    Initialize with a specific unit string like this: 
+        u = MeasurementUnit('cm')
+
+    In this case the prefix is 'c' (centi), and the suffix is 'm' (metres):
+    
+        assert(MeasurementUnit('centimetres') == u)
+        
+    To get the canonical (i.e. 'mm') representation of 1 cm:
+    
+        u.to_canon(1)
+        
+    This returns `10.0`.
+    
+    Attributes
+    -----------
+
+    prefix: str
+        The original prefix
+    suffix: str
+        The original suffix
+    unit_string: str
+        The original string    
+    canonical_prefix: str
+        The canonical form for the prefix.
+    canonical_suffix: str
+        The canonical form for the suffix.
+    canonical_unit_string: str
+        canonical_prefix + canonical_suffix
+    all_suffixes: list
+        All available suffixes
+    all_prefixes: list
+        All available prefixes
+
+    Methods
+    ------------
+    __init__
+    __eq__
+    __ne__
+    to_canon
+        transforms v from original units to canonical units
+    from_canon
+        the inverse of to_canon
+
+    """    
+    
+    # PREFIXES
+    SI_prefixes = {'': 1,
+                   'c': 1e-2, 'centi': 1e-2,
+                   'm': 1e-3, 'milli': 1e-3,
+                   # Two ways to get the Greek Mu character in unicode:
+                   'u': 1e-6, '\u00B5': 1e-6, '\u03BC': 1e-6, 'micro': 1e-6,
+                   'n': 1e-9, 'nano': 1e-9,
+                   'k': 1e+3, 'K': 1e+3, 'kilo': 1e+3,
```

Comment:
`K` is not a valid SI prefix (it is not a valid substitute for `k`). Also, it conflicts with Kelvin if we're using K for that.